ContactManager/ContactManagerUI/ViewController/AddContactViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  AddContactViewController.swift
+//  ContactManagerUI
+//
+//  Created by 신동오 on 2023/02/06.
+//
+
+import UIKit
+
+protocol AddContactViewControllerDelegate {
+    func didContactChanged()
+}
+
+final class AddContactViewController: UIViewController {
+    // MARK: - Properties
+    var delegate: AddContactViewControllerDelegate?
+    
+    @IBOutlet private weak var nameTextField: UITextField!
+    @IBOutlet private weak var ageTextField: UITextField!
+    @IBOutlet private weak var phoneNumberTextField: UITextField!
+    @IBOutlet private weak var navigationBar: UINavigationBar!
+
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationBar.shadowImage = UIImage()
+        configureTextField()
+    }
+    
+    // MARK: - Actions
+    @IBAction private func tappedCancelButton(_ sender: UIBarButtonItem) {
+        showCheckCancelAlert()
+    }
+    
+    @IBAction private func tappedSaveButton(_ sender: UIBarButtonItem) {
+        guard let name = nameTextField.text ,
+              let age = ageTextField.text,
+              let phoneNumber = phoneNumberTextField.text else { return }
+        
+        let inputArray = [name, age, phoneNumber]
+        
+        do {
+            let contact = try InputManager.contact(from: inputArray)
+            ContactManager.shared.add(contact: contact)
+        } catch {
+            showFailAlert(withTitle: error.localizedDescription)
+            return
+        }
+        
+        dismiss(animated: true)
+        delegate?.didContactChanged()
```

Comment:
dismiss 하고나서 메소드 호출을 하는 것이 목적이라면 completion을 써도 좋을 것 같아요. dismiss 후에 호출되는 것이 100% 보장되니까요. 

```suggestion
        dismiss(animated: true, completion: {
            self.delegate?.didContactChanged()
        })
```

Reply:
dismiss 에도 completion 이 있다는 사실을 덕분에 알게 되었습니다 :)
dismiss 이후 동작 수행 시 메모리에서 해제되지 않으리라는 보장이 없다는 부분을 덕분에 생각하게 되었습니다.
감사합니다!
> 해당 커밋: 5bd7e85